src/devices/src/virtio/net/unixgram.rs

```diff
@@ -37,7 +42,9 @@ impl Gvproxy {
         // allows the server to remove the socket after the connection.
         connect(fd, &peer_addr).map_err(ConnectError::Binding)?;
 
-        send(fd, &VFKIT_MAGIC, MsgFlags::empty()).map_err(ConnectError::SendingMagic)?;
+        if vfkit_magic {
+            send(fd, &VFKIT_MAGIC, MsgFlags::empty()).map_err(ConnectError::SendingMagic)?;
+        }
```

Comment:
Nice. Maybe rename to send_vfkit_magic to make it more clear.

Reply:
Renamed, thanks.